[quant][graphmode][fx] Support preserving attributes in deepcopy of observed/quantized graphmodule

jerryzh168 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#56550 [quant][graphmode][fx] Support preserving attributes in deepcopy of observed/quantized graphmodule**

Summary:
Add support for preserving a list of attributes on observed/quantized GraphModule

Test Plan:
python test/test_quantization.py TestQuantizeFx.test_deepcopy_preserve_attributes

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D27899317](https://our.internmc.facebook.com/intern/diff/D27899317)

jerryzh168 committed:
[quant][graphmode][fx] Support preserving attributes in deepcopy of observed/quantized graphmodule

Summary:
Add support for preserving a list of attributes on observed/quantized GraphModule

Test Plan:
python test/test_quantization.py TestQuantizeFx.test_deepcopy_preserve_attributes

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

t10-13rocket reviewed (approved):
LGTM

t10-13rocket reviewed (approved):
LGTM

jerryzh168 committed:
Update on "[quant][graphmode][fx] Support preserving attributes in deepcopy of observed/quantized graphmodule"


Summary:
Add support for preserving a list of attributes on observed/quantized GraphModule

Test Plan:
python test/test_quantization.py TestQuantizeFx.test_deepcopy_preserve_attributes

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D27899317](https://our.internmc.facebook.com/intern/diff/D27899317)

[ghstack-poisoned]

vkuzo commented on torch/quantization/quantize_fx.py:
```diff
@@ -233,6 +236,11 @@ def fuse_fx(model: torch.nn.Module,
     torch._C._log_api_usage_once("quantization_api.quantize_fx.fuse_fx")
     assert not model.training, 'fuse_fx only works on models in eval mode'
     graph_module = torch.fx.symbolic_trace(model)  # type: ignore
+    preservd_attributes: Set[str] = set()
```
should this be `preserved_attributes` (typo)?

vkuzo replied:
can we update the docblock of `fuse_fx` and `prepare_fx` with this?

jerryzh168 committed:
Update on "[quant][graphmode][fx] Support preserving attributes in deepcopy of observed/quantized graphmodule"


Summary:
Add support for preserving a list of attributes on observed/quantized GraphModule

Test Plan:
python test/test_quantization.py TestQuantizeFx.test_deepcopy_preserve_attributes

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D27899317](https://our.internmc.facebook.com/intern/diff/D27899317)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][fx] Support preserving attributes in deepcopy of observed/quantized graphmodule"


Summary:
Add support for preserving a list of attributes on observed/quantized GraphModule

Test Plan:
python test/test_quantization.py TestQuantizeFx.test_deepcopy_preserve_attributes

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D27899317](https://our.internmc.facebook.com/intern/diff/D27899317)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][fx] Support preserving attributes in deepcopy of observed/quantized graphmodule"


Summary:
Add support for preserving a list of attributes on observed/quantized GraphModule

Test Plan:
python test/test_quantization.py TestQuantizeFx.test_deepcopy_preserve_attributes

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D27899317](https://our.internmc.facebook.com/intern/diff/D27899317)

[ghstack-poisoned]
